misk/src/main/kotlin/misk/eventrouter/ConsistentHashing.kt

```diff
@@ -0,0 +1,36 @@
+package misk.eventrouter
+
+import com.google.common.hash.Hashing
+
+class ConsistentHashing : ClusterMapper {
+  private val hashFunction = Hashing.murmur3_128()
+  private val mod = 65536L
+  private val virtualPoints = 16
+
+  // TODO(tso): cache for each clusterSnapshot
+  override fun topicToHost(clusterSnapshot: ClusterSnapshot, topic: String): String {
+    val hosts = clusterSnapshot.hosts.sorted()
+
+    val topicHash = Math.floorMod(hashFunction.hashString(topic, Charsets.UTF_8).padToLong(), mod)
```

Comment:
I used mod because negative numbers were annoying and it enforces a set number of bits (2^16 in this case, rather than the size of a long)

Maybe it doesn't really matter